lib/modules/swarm/upload.js

```diff
@@ -9,46 +9,50 @@ class Swarm {
   }
 
   deploy() {
-    let self = this;
-    async.waterfall([
-      function findBinary(callback) {
-        let swarm_bin = shelljs.which('swarm');
-
-        if (swarm_bin === 'swarm not found' || !swarm_bin) {
-          console.log('=== WARNING: Swarm not in an executable path. Guessing ~/go/bin/swarm for path'.yellow);
-          swarm_bin = "~/go/bin/swarm";
+    return new Promise((resolve, reject) => {
+      let self = this;
+      async.waterfall([
+        function findBinary(callback) {
+          let swarm_bin = shelljs.which('swarm');
+
+          if (swarm_bin === 'swarm not found' || !swarm_bin) {
+            console.log('=== WARNING: Swarm not in an executable path. Guessing ~/go/bin/swarm for path'.yellow);
+            swarm_bin = "~/go/bin/swarm";
+          }
+
+          return callback(null, swarm_bin);
+        },
+        function runCommand(swarm_bin, callback) {
+          let cmd = `"${swarm_bin}" --defaultpath ${self.buildDir} index.html --recursive up ${self.buildDir}`;
+          console.log(("=== adding " + self.buildDir + " to swarm").green);
+          console.log(cmd.green);
+          let result = shelljs.exec(cmd);
```

Comment:
Same comments as for upload above. No need to return a last callback. Also, use async exec.